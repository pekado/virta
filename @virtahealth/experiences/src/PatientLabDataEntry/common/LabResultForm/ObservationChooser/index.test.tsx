import * as React from "react";
import { act, render, waitFor, fireEvent } from "@testing-library/react-native";
import {
  createComponentWithVirtaContext,
  WrapComponentWithVirtaContext,
} from "../../../../test";
import { hbA1cQuestionnaire } from "../../../questionnaires";
import { ObservationChooser } from "./index";

const ChooserTester = () => {
  const [questionnaires, setQuestionnaires] = React.useState<
    fhir.Questionnaire[]
  >([]);
  const [responses, setResponses] = React.useState<
    fhir.QuestionnaireResponse[]
  >([]);

  return (
    <ObservationChooser
      observationQuestionnaires={questionnaires}
      setObservationQuestionnaires={setQuestionnaires}
      observationResponses={responses}
      setObservationResponses={setResponses}
    />
  );
};

describe("ObservationChooser", () => {
  it("renders", async () => {
    const observationChooser = createComponentWithVirtaContext(
      <ChooserTester />
    );
    await waitFor(() => {
      observationChooser.getByTestId("lipid");
    });
    expect(observationChooser).toMatchSnapshot();
  });

  describe("test questionnaire displays", () => {
    it.each([
      ["a1c", "HbA1c"],
      ["metabolicPanel", "Metabolic panel"],
      ["lipid", "Lipids"],
      ["antiGad", "GAD65 Antibody"],
      ["cPeptide", "C peptide panel"],
      ["hsCrp", "hsCRP"],
      ["albuminCreatinine", "Albumin/Creatinine Ratio, Urine"],
      ["nmrLipoprofile", "Lipid NMR panel"],
      ["serumInsulin", "Insulin, Serum"],
      ["tsh", "TSH"],
      ["clinicalTrial", "Clinical Trial"],
      ["biometric", "Biometric"],
    ])(
      "displays %s when the %s checkbox is selected",
      async (checkboxTestId, questionnaireDisplayText) => {
        let questionnaires: fhir.Questionnaire[] = [];
        const setQuestionnaires = jest.fn(
          (data: fhir.Questionnaire[]) => (questionnaires = [...data])
        );
        const responses: fhir.QuestionnaireResponse[] = [];
        const setResponses = jest.fn();

        const observationChooser = render(
          <WrapComponentWithVirtaContext>
            <ObservationChooser
              observationQuestionnaires={questionnaires}
              setObservationQuestionnaires={setQuestionnaires}
              observationResponses={responses}
              setObservationResponses={setResponses}
            />
          </WrapComponentWithVirtaContext>
        );

        const inputCheckbox = observationChooser.getByTestId(checkboxTestId);
        act(() => fireEvent.press(inputCheckbox));
        observationChooser.update(
          <WrapComponentWithVirtaContext>
            <ObservationChooser
              observationQuestionnaires={questionnaires}
              setObservationQuestionnaires={setQuestionnaires}
              observationResponses={responses}
              setObservationResponses={setResponses}
            />
          </WrapComponentWithVirtaContext>
        );

        // TODO - this should work but doesn't due to a bug
        // https://github.com/callstack/react-native-testing-library/issues/553
        // once the bug is fixed, switch to the original version of the test and remove the if/else
        // await waitFor(() =>
        //   observationChooser.getAllByText(questionnaireDisplayText)
        // );
        // TODO: Jest rules don't like conditional expects...
        /* eslint-disable jest/no-conditional-expect */
        if (
          checkboxTestId === "a1c" ||
          checkboxTestId === "antiGad" ||
          checkboxTestId === "hsCrp" ||
          checkboxTestId === "tsh"
        ) {
          expect(
            observationChooser.toJSON()[3]?.children[0].children[1].children[0]
              .children[0].children[0].children[0].children[0]
          ).toEqual(questionnaireDisplayText);
        } else {
          expect(
            observationChooser.toJSON()[3]?.children[0].children[0].children[0]
          ).toEqual(questionnaireDisplayText);
        }
        /* eslint-enable jest/no-conditional-expect */
      }
    );
  });

  it("displays multiple questionnaires", async () => {
    let questionnaires: fhir.Questionnaire[] = [];
    const setQuestionnaires = jest.fn(
      (data: fhir.Questionnaire[]) => (questionnaires = [...data])
    );
    const responses: fhir.QuestionnaireResponse[] = [];
    const setResponses = jest.fn();

    const observationChooser = render(
      <WrapComponentWithVirtaContext>
        <ObservationChooser
          observationQuestionnaires={questionnaires}
          setObservationQuestionnaires={setQuestionnaires}
          observationResponses={responses}
          setObservationResponses={setResponses}
        />
      </WrapComponentWithVirtaContext>
    );
    const a1cCheckbox = observationChooser.getByTestId("a1c");
    act(() => fireEvent.press(a1cCheckbox));
    observationChooser.update(
      <WrapComponentWithVirtaContext>
        <ObservationChooser
          observationQuestionnaires={questionnaires}
          setObservationQuestionnaires={setQuestionnaires}
          observationResponses={responses}
          setObservationResponses={setResponses}
        />
      </WrapComponentWithVirtaContext>
    );

    // TODO - this should work but doesn't due to a bug
    // https://github.com/callstack/react-native-testing-library/issues/553
    // once the bug is fixed, switch to the original version of the test
    // await waitFor(() => observationChooser.getAllByText("Hgb A1c"));
    expect(
      String(
        observationChooser.toJSON()[3]?.children[0].children[1].children[0]
          .children[0].children[0].children[0].children[0]
      )
    ).toEqual("HbA1c");

    const lipidCheckbox = observationChooser.getByTestId("lipid");
    act(() => fireEvent.press(lipidCheckbox));
    observationChooser.update(
      <WrapComponentWithVirtaContext>
        <ObservationChooser
          observationQuestionnaires={questionnaires}
          setObservationQuestionnaires={setQuestionnaires}
          observationResponses={responses}
          setObservationResponses={setResponses}
        />
      </WrapComponentWithVirtaContext>
    );

    // TODO - this should work but doesn't due to a bug
    // https://github.com/callstack/react-native-testing-library/issues/553
    // once the bug is fixed, switch to the original version of the test
    // await waitFor(() => observationChooser.getByText("Lipid 1996 panel"));
    expect(
      observationChooser.toJSON()[4]?.children[0].children[0].children[0]
    ).toEqual("Lipids");
  });

  describe("questionnaire responses", () => {
    it("sets the response for a questionnaire", () => {
      const mockResponseSetter = jest.fn();
      const observationChooser = createComponentWithVirtaContext(
        <ObservationChooser
          observationQuestionnaires={[hbA1cQuestionnaire]}
          setObservationQuestionnaires={() => undefined}
          observationResponses={[
            {
              item: [
                {
                  item: [],
                  linkId: "/4548-4",
                },
              ],
              status: "in-progress",
            },
          ]}
          setObservationResponses={mockResponseSetter}
        />
      );
      const hbA1cInput = observationChooser.getByTestId("/4548-4");
      fireEvent.changeText(hbA1cInput, "6.5");
      expect(mockResponseSetter).toHaveBeenCalledWith([
        {
          status: "in-progress",
          item: [
            {
              linkId: "/4548-4",
              item: [
                {
                  linkId: "/4548-4",
                  answer: [
                    {
                      valueDecimal: "6.5",
                    },
                  ],
                },
              ],
            },
          ],
        },
      ]);
    });
  });
});
