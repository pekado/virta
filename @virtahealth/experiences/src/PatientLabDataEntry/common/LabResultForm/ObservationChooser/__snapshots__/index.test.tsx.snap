// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ObservationChooser renders 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "rgb(36, 38, 43)",
          "fontFamily": "Whitney-Book",
          "fontSize": 12,
          "fontWeight": "500",
          "letterSpacing": 0,
          "lineHeight": 12,
        },
        Object {
          "fontWeight": "600",
          "paddingBottom": 12,
        },
      ]
    }
  >
    Tests Ordered
  </Text>,
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        },
        Object {
          "flexWrap": "wrap",
        },
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="a1c"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="a1c-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                A1C
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="metabolicPanel"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="metabolicPanel-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Metabolic Panel
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="lipid"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="lipid-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Lipids
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="antiGad"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="antiGad-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                GAD65 Antibody
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="cPeptide"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="cPeptide-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                C-Peptide
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="hsCrp"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="hsCrp-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                hsCRP
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="albuminCreatinine"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="albuminCreatinine-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Albumin/Creatinine, Urine
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="nmrLipoprofile"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="nmrLipoprofile-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Lipid NMR
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="serumInsulin"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="serumInsulin-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Serum Insulin
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="tsh"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="tsh-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                TSH
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="clinicalTrial"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="clinicalTrial-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Clinical Trial
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
    <View>
      <View
        style={
          Array [
            Object {
              "display": "flex",
              "flexDirection": "column",
            },
          ]
        }
      >
        <TouchableOpacity
          accessibilityLabel="checkbox"
          onPress={[Function]}
          testID="biometric"
        >
          <View
            hasDescription={false}
            isChecked={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "rgb(214, 217, 221)",
                  "borderRadius": 0,
                  "borderWidth": 0,
                  "display": "flex",
                  "flexDirection": "row",
                  "marginRight": 8,
                  "paddingBottom": 0,
                  "paddingLeft": 0,
                  "paddingRight": 0,
                  "paddingTop": 0,
                },
              ]
            }
          >
            <View
              isChecked={false}
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "white",
                    "borderColor": "rgb(214, 217, 221)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "height": 16,
                    "justifyContent": "center",
                    "marginRight": 8,
                    "width": 16,
                  },
                ]
              }
              testID="biometric-box"
            />
            <View
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "column",
                    "width": "100%",
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(36, 38, 43)",
                      "fontFamily": "Whitney-Book",
                      "fontSize": 12,
                      "fontWeight": "500",
                      "letterSpacing": 0,
                      "lineHeight": 12,
                    },
                    Object {
                      "fontSize": 16,
                      "fontWeight": "300",
                      "lineHeight": 16,
                    },
                  ]
                }
              >
                Biometric
              </Text>
            </View>
          </View>
        </TouchableOpacity>
      </View>
      <View
        height={12}
        style={
          Array [
            Object {
              "height": 12,
              "width": 0,
            },
          ]
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "height": 0,
            "width": 8,
          },
        ]
      }
      width={8}
    />
  </View>,
  <View
    height={10}
    style={
      Array [
        Object {
          "height": 10,
          "width": 0,
        },
      ]
    }
  />,
]
`;
